context/src/main/java/io/micronaut/runtime/server/GracefulShutdownManager.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.runtime.server;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.NonNull;
+import jakarta.inject.Singleton;
+
+import java.util.List;
+import java.util.OptionalLong;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Helper class that can be used to call all {@link GracefulShutdownCapable} beans.
+ *
+ * @since 4.9.0
+ * @author Jonas Konrad
+ */
+@Singleton
+@Requires(classes = GracefulShutdownCapable.class)
+public final class GracefulShutdownManager {
```

Comment:
since the constructor is not public make `non-public` and annotate with `@Internal`

Reply:
I think it's useful to be kept as API. The constructor is non-public so we can inject more parameters in the future without issue.

Reply:
How about Experimental?

Reply:
yes add `@Experimental`